src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,376 @@
+   * @default 1
+   */
+  secondStep?: number;
+};
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+  const [focusedTime, setFocusedTime] = React.useState(
+    selectedTime ?? new Date(),
+  );
+  const [period, setPeriod] = React.useState<'PM' | 'AM' | undefined>(
+    use12Hours ? (focusedTime?.getHours() > 11 ? 'PM' : 'AM') : undefined,
+  );
+
+  React.useEffect(() => {
+    setFocusedTime(date ?? new Date());
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedHour = use12Hours ? formatHourFrom12(hour, period) : hour;
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setHours(adjustedHour);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setMinutes(minutes);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setSeconds(seconds);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onPeriodClick = (value: string) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    const currentHours = adjustedSelectedTime.getHours();
+    if (value === 'AM' && currentHours > 11) {
+      setPeriod(value);
+      adjustedSelectedTime.setHours(currentHours - 12);
+    }
+    if (value === 'PM' && currentHours <= 12) {
+      setPeriod(value);
+      adjustedSelectedTime.setHours(currentHours + 12);
+    }
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const updateCurrentTime = (time: Date) => {
+    setFocusedTime(time);
+    setSelectedTime(time);
+    onChange?.(time);
+  };
+
+  const onHourFocus = (hour: number | undefined) => {
+    const adjustedHour =
+      use12Hours && hour ? formatHourFrom12(hour, period) : hour;
+    applyFocus(adjustedHour, (value, date) => date.setHours(value));
+  };
+
+  const onMinuteFocus = (minute: number | undefined) => {
+    applyFocus(minute, (value, date) => date.setMinutes(value));
+  };
+
+  const onSecondFocus = (second: number | undefined) => {
+    applyFocus(second, (value, date) => date.setSeconds(value));
+  };
+
+  const applyFocus = (
+    value: number | undefined,
+    callback: (value: number, date: Date) => void,
+  ) => {
+    if (!!value) {
+      const date = new Date(focusedTime);
+      callback(value, date);
+      setFocusedTime(date);
+    } else {
```

Comment:
When else is called?

Reply:
Ok I found it. It is triggered when focusing item with 0 and it causes focus to jump to currently seelected value or if nothing is selected to the current time value.

Reply:
Yup, it is not needed as you always get a value. So you can change `value` type to just `number` also the same `on...Focus`

Reply:
Not always.. onBlur I send undefined.. It is this hack to get out from the focus..
Ill try to check again if there is a way without it..

Reply:
removed